api-admin/src/main/java/bio/terra/pearl/api/admin/controller/StudyEnvironmentController.java

```diff
@@ -68,17 +68,47 @@ public ResponseEntity<Object> getKitTypes(
       String portalShortcode, String studyShortcode, String envName) {
     AdminUser adminUser = requestService.requireAdminUser(request);
     EnvironmentName environmentName = EnvironmentName.valueOfCaseInsensitive(envName);
-    List<KitType> kitTypes =
-        studyEnvExtService.getKitTypes(adminUser, portalShortcode, studyShortcode, environmentName);
+    PortalStudyEnvAuthContext authContext =
+        PortalStudyEnvAuthContext.of(adminUser, portalShortcode, studyShortcode, environmentName);
+
+    List<KitType> kitTypes = studyEnvExtService.getKitTypes(authContext);
+    return ResponseEntity.ok(kitTypes);
+  }
+
+  @Override
+  public ResponseEntity<Object> patchKitTypes(
+      String portalShortcode, String studyShortcode, String envName, Object body) {
+    AdminUser operator = requestService.requireAdminUser(request);
+    EnvironmentName environmentName = EnvironmentName.valueOfCaseInsensitive(envName);
+    PortalStudyEnvAuthContext authContext =
+        PortalStudyEnvAuthContext.of(operator, portalShortcode, studyShortcode, environmentName);
+
+    List<String> kitTypeNames =
+        objectMapper.convertValue(
+            body, objectMapper.getTypeFactory().constructCollectionType(List.class, String.class));
```

Comment:
TIL about `constructCollectionType`, that's handy to know

Reply:
I also hadn't seen this before, nifty.

But unless there's an advantage to doing it this way, let's stick with using `TypeReference`, which is how we do it in other controllers (see e.g. MailingListController:51.  (IMHO the syntax is a little easier to parse)